molecularnodes/entities/molecule/array.py

```diff
@@ -0,0 +1,52 @@
+import uuid
+from biotite.structure import AtomArray
+
+from .base import Molecule
+from .pdb import _comp_secondary_structure
+from ..base import EntityType
+
+
+class Array(Molecule):
+    def __init__(self, array: AtomArray, name="AtomArray"):
+        self.file_path = None
+        self.file = "ARRAY_LOADED_DIRECTLY"
+        self._frames_collection = None
+        self._entity_type = EntityType.MOLECULE
+        self._assemblies = lambda: None
+        self._uuid = str(uuid.uuid4())
+        self.array = self._validate_structure(array)
+        self.n_atoms = self.array.array_length()    # note: I am not sure if this is consistent
+        self.create_object(name=name)               # Create the Blender object
+
+    def read(self, file_path):
+        pass
+
+    def _validate_structure(self, array: AtomArray):
+        # TODO: implement entity ID, sec_struct for PDB files
```

Comment:
A complete implementation should handle these fields as well by setting defaults.  The assumption here is that the Array is being passed and the file is no longer available. I letft it unimplemented pending a good solution